lua/lspconfig/server_configurations/tinymist.lua

```diff
@@ -0,0 +1,16 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'tinymist' },
+    filetypes = { 'typst' },
+    root_dir = util.find_git_ancestor,
+    single_file_support = false,
```

Comment:
```suggestion
    single_file_support = true,
```

Reply:
Theoretically, this lsp does not have single file support...

Reply:
Well, actually it’s not that it can’t run on a single file, it just has an incomplete experience.

Reply:
okay .but in there this also mean use cwd as root . so i wonder this not work ?

Reply:
If you use cwd as root, it works, but if you don't carry root initially, it may take time to initialize and cause lagging
![21c683246004b17b040c3403148c564c](https://github.com/neovim/nvim-lspconfig/assets/31800073/9b98cb66-51bb-4ea2-a6f0-1fd27a3fae53)

Reply:
You can revise it and make the final decision, it's my first time to do this kind of pr, please forgive me if there may be some wrong.

Reply:
> but in there this also mean use cwd as root . so i wonder this not work ?

I don't really understand `single_file_support`, but if neovim as a client will somehow provide a root_dir, the lsp will be not problem at all. They we can go with `single_file_support = true`, and let people report issues if any.

Reply:
some code here , you can find in configs.lua

```lua
        elseif config.single_file_support then
          -- This allows on_new_config to use the parent directory of the file
          -- Effectively this is the root from lspconfig's perspective, as we use
          -- this to attach additional files in the same parent folder to the same server.
          -- We just no longer send rootDirectory or workspaceFolders during initialization.
          if not api.nvim_buf_is_valid(bufnr) or (#bufname ~= 0 and not util.bufname_valid(bufname)) then
            return
          end
          local pseudo_root = #bufname == 0 and pwd or util.path.dirname(util.path.sanitize(bufname))
          M.manager:add(pseudo_root, true, bufnr)
        end
```